applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_kinematic_plasticity.cpp

```diff
@@ -0,0 +1,496 @@
+        return BaseType::GetValue(rThisVariable, rValue);
+    }
+
+    return rValue;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+Matrix& GenericSmallStrainKinematicPlasticity<TConstLawIntegratorType>::GetValue(
+    const Variable<Matrix>& rThisVariable,
+    Matrix& rValue
+    )
+{
+    if (rThisVariable == PLASTIC_STRAIN_TENSOR) {
+        rValue = MathUtils<double>::StrainVectorToTensor(mPlasticStrain);
+    } else {
+        return BaseType::GetValue(rThisVariable, rValue);
+    }
+
+    return rValue;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+double& GenericSmallStrainKinematicPlasticity<TConstLawIntegratorType>::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<double>& rThisVariable,
+    double& rValue
+    )
+{
+    if (rThisVariable == EQUIVALENT_PLASTIC_STRAIN) {
+        const Vector& r_stress_vector = rParameterValues.GetStressVector();
+        TConstLawIntegratorType::CalculateEquivalentPlasticStrain(r_stress_vector, mUniaxialStress, mPlasticStrain, 0.0, rParameterValues, rValue);
+
+        return rValue;
+    } else {
+        return this->GetValue(rThisVariable, rValue);
+    }
+
+    return rValue;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+Vector& GenericSmallStrainKinematicPlasticity<TConstLawIntegratorType>::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<Vector>& rThisVariable,
+    Vector& rValue
+    )
+{
+    return BaseType::CalculateValue(rParameterValues, rThisVariable, rValue);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+Matrix& GenericSmallStrainKinematicPlasticity<TConstLawIntegratorType>::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<Matrix>& rThisVariable,
+    Matrix& rValue
+    )
+{
+    if (rThisVariable == INTEGRATED_STRESS_TENSOR) {
+        //1.-Compute total deformation gradient
+        const Matrix& DeformationGradientF = rParameterValues.GetDeformationGradientF();
+        //2.-Right Cauchy-Green tensor C
+        const Matrix right_cauchy_green = prod(trans(DeformationGradientF), DeformationGradientF);
+        Vector strain_vector = ZeroVector(6);
+
+        //E= 0.5*(FT*F-1) or E = 0.5*(C-1)
+        strain_vector[0] = 0.5 * (right_cauchy_green(0, 0) - 1.00);

```

Comment:
Look in the constituve_law_utilities. There is an implementation of this

Reply:
I'll remove thx

Reply:
solved